src/full/Agda/Utils/TypeLevel.hs

```diff
@@ -72,6 +72,17 @@ type family Constant (b :: l) (as :: [k]) :: [l] where
 type Arrows   (as :: [Type]) (r :: Type) = Foldr (->) r as
 type Products (as :: [Type])             = Foldr (,) () as
 
+data StrictPair a b = Pair a b
+type StrictProducts (as :: [Type]) = Foldr StrictPair () as
+
+strictCurry :: (StrictPair a b -> c) -> (a -> b -> c)
+strictCurry f = \ !a !b -> f (Pair a b)
+{-# inline strictCurry #-}
```

Comment:
Since you did it for the other files already, please also uppercase INLINE in this file.